From ef4f0b3b23ecd3ea6719cc94c1a62e0ea9fd4e94 Mon Sep 17 00:00:00 2001
From: dyphire <qimoge@gmail.com>
Date: Thu, 21 May 2026 12:04:37 +0800
Subject: [PATCH] libplacebo: add patch

---
 packages/libplacebo.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/packages/libplacebo.cmake b/packages/libplacebo.cmake
index b5797dc..89c2c8c 100644
--- a/packages/libplacebo.cmake
+++ b/packages/libplacebo.cmake
@@ -13,6 +13,8 @@ ExternalProject_Add(libplacebo
     SOURCE_DIR ${SOURCE_LOCATION}
     GIT_CLONE_FLAGS "--filter=tree:0"
     GIT_SUBMODULES ""
+    PATCH_COMMAND ${EXEC} curl -sL https://code.videolan.org/videolan/libplacebo/-/merge_requests/850.patch | git am --3way --whitespace=fix
+          COMMAND ${EXEC} curl -sL https://code.videolan.org/videolan/libplacebo/-/merge_requests/852.patch | git am --3way --whitespace=fix
     UPDATE_COMMAND ""
     CONFIGURE_COMMAND ""
     COMMAND bash -c "rm -rf <SOURCE_DIR>/3rdparty/glad"
-- 
2.47.0.windows.1

