diff --git a/libavcodec/apv_parser.c b/libavcodec/apv_parser.c
--- a/libavcodec/apv_parser.c
+++ b/libavcodec/apv_parser.c
@@ -194,7 +194,7 @@ static av_cold int init(AVCodecParserContext *s)
     return 0;
 }
 
-static av_cold void close(AVCodecParserContext *s)
+static av_cold void apv_parser_close(AVCodecParserContext *s)
 {
     APVParseContext *p = s->priv_data;
     ParseContext *pc = &p->pc;
@@ -211,5 +211,5 @@ const FFCodecParser ff_apv_parser = {
     .priv_data_size = sizeof(APVParseContext),
     .init           = init,
     .parse          = parse,
-    .close          = close,
+    .close          = apv_parser_close,
 };
