diff --git a/src/base/md5.h b/src/base/md5.h
--- a/src/base/md5.h
+++ b/src/base/md5.h
@@ -43,9 +43,13 @@ typedef struct {
     MD5_u32plus block[16];
 } MD5_CTX;
 
+#define MD5_Init aribcc_md5_init
+#define MD5_Update aribcc_md5_update
+#define MD5_Final aribcc_md5_final
+
 extern void MD5_Init(MD5_CTX *ctx);
 extern void MD5_Update(MD5_CTX *ctx, const void *data, unsigned long size);
 extern void MD5_Final(unsigned char *result, MD5_CTX *ctx);
 
 
 #ifdef __cplusplus
