diff --git a/al/auxeffectslot.cpp b/al/auxeffectslot.cpp
index 77cfa5793..71fe17715 100644
--- a/al/auxeffectslot.cpp
+++ b/al/auxeffectslot.cpp
@@ -211,7 +211,7 @@ void AddActiveEffectSlots(std::span<gsl::not_null<al::EffectSlot*> const> const
         newarray = EffectSlotBase::CreatePtrArray(newcount<<1);
         std::ranges::copy(*oldarray | std::views::take(newcount), newarray->begin());
     }
-    std::ranges::fill(*newarray | std::views::drop(newcount), nullptr);
+    std::ranges::fill(std::span{*newarray}.subspan(newcount), nullptr);

     auto oldarray = context->mActiveAuxSlots.exchange(std::move(newarray),
         std::memory_order_acq_rel);
diff --git a/alc/alc.cpp b/alc/alc.cpp
--- a/alc/alc.cpp
+++ b/alc/alc.cpp
@@ -1781,7 +1781,8 @@
          * configuration in aluInitEffectPanning.
          */
         std::ranges::for_each(context->mEffectSlotClusters
-            | std::views::transform(&ContextBase::EffectSlotCluster::operator*)
+            | std::views::transform([](ContextBase::EffectSlotCluster const &cluster) -> auto&
+            { return *cluster; })
             | std::views::join, [](EffectSlotBase &slot)
         {
             slot.mWetBuffer.clear();
