diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -404,6 +404,11 @@

     if(ALSOFT_WERROR)
         list(APPEND C_FLAGS -Werror)
+    elseif(APPLE)
+        list(APPEND C_FLAGS -Werror=format-security
+            $<$<COMPILE_LANGUAGE:CXX>:-Wno-elaborated-enum-base>
+            $<$<COMPILE_LANGUAGE:C>:-Werror=implicit -Werror=incompatible-pointer-types
+                -Werror=int-conversion>)
     else()
         list(APPEND C_FLAGS -Werror=undef -Werror=format-security
             $<$<COMPILE_LANGUAGE:C>:-Werror=implicit -Werror=incompatible-pointer-types
